src/selectors/identity.js

```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
 
 export const selectShowEmptyClaimCategories = (state) => state.identity.getIn(['personalInformation', 'showEmptyClaimCategories'], false);
 
+export const selectClaims = (state) => state.identity.getIn(['personalInformation', 'claims', 'byId'], IMap());
+
 export const selectClaimCategoriesToDisplay = createSelector(
-  [selectClaimCategories, selectShowEmptyClaimCategories],
-  (claimCategories, showEmptyClaimCategories) => {
+  [selectClaimCategories, selectShowEmptyClaimCategories, selectClaims],
+  (claimCategories, showEmptyClaimCategories, claims) => {
     if (!showEmptyClaimCategories) {
-      return claimCategories.filter((claimCategory) => claimCategory.get('claims', IList()).size > 0);
+      return claimCategories.filter((claimCategory) => claims.some((claim) => claim.get('categoryId', '') === claimCategory.get('id', '')));
     }
     return claimCategories;
   },
 );
 
-const selectClaims = (state) => state.identity.getIn(['personalInformation', 'claims', 'byId'], IMap());
+export const selectClaimsObj = (state) => state.identity.getIn(['personalInformation', 'claims'], IMap());
```

Comment:
Think about a more descriptive name, maybe something like selectClaimsReducerState or something down that line.
Obj is just to ambiguous.